starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/MicronautBuildCache.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.build.gradle;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class MicronautBuildCache extends GradleBuildCache {
+
+    public MicronautBuildCache(@Nullable BuildCacheCredentialsProvider buildCacheCredentialsProvider) {
+        super(buildCacheCredentialsProvider);
+    }
+
+    @Override
+    @NonNull
+    public String getName() {
+        return "micronaut-build-cache";
+    }
+
+    @Override
+    public String getTitle() {
+        return "Micronaut Build Cache";
+    }
+
+    @Override
+    public boolean isVisible() {
+        return false;
+    }
+
+    @Override
+    @NonNull
+    public String getUrl() {
+        return "https://ge.micronaut.io/cache/";
+    }
+
+    @Override
+    public boolean push() {
+        return true;
```

Comment:
Is this for CI?  I assume we only want to push from CI?  Is this a sensible default?  Or should we default to some property that only exists on CI?

Reply:
Yes, it would be used in `CI`. Please, note that `MicronautBuildCache` is not `visible` see `isVisible() { return false; }`. It would be used by our generated projects on Ci. 

I think we needed as `push = true` according to https://github.com/micronaut-projects/micronaut-starter/pull/1019/files See `maybeaddGradleBuldCacheGradleConfiguration`method. Maybe @melix can clarify this.

Reply:
Cool!  I was just worried people might push from their local machines, but I guess without credentials it would fail 👍